### PR TITLE
Make terraform error output available in termination log

### LINKED ISSUE #67: Make terraform error output available via `terminationMessagePath`
**How to categorize this issue?**

/area quality usability
/kind enhancement
/priority normal

**What would you like to be added**:

Terraformer should copy terraform's error output on unsuccessful operations to a file, that can be set as the container's `terminationMessagePath` (default `/dev/termination-log`), see [this doc](https://kubernetes.io/docs/tasks/debug-application-cluster/determine-reason-pod-failure/#customizing-the-termination-message).
This will populate the error message in `PodStatus.containerStatuses[].state.terminated.message`.

The limitation of `4KiB` per container / `12KiB` per Pod shouldn't be a problem here. `4KiB` should be enough to transport the last few relevant lines of terraforms out, which are typically the most relevant.

**Why is this needed**:

By only copying the terraform error output, it will be easier for extensions to retrieve the error logs for analysis / error code detection.
With `v2` the log volume apart from terraform's output significantly increased (especially when using `debug` log level) and they are written in `json` format by default.
When such erroneous logs are transported to the Shoot status for example, it results in a pretty unreadable error message, which is a bad user experience.

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area usability
/kind enhancement

**What this PR does / why we need it**:

Make terraform error output available in termination log as described in #67.

Also
- disable terraform colorized output 
- copy terraformer binary after tf and plugins (for faster subsequent pushes during development)

**Which issue(s) this PR fixes**:
Fixes #67

**Special notes for your reviewer**:

/keep-commits

In draft until I've made adaptions in `g/g/ext/terraformer` and tested this end-to-end with `g/aws`.

**Release note**:

```feature operator
Terraformer now copies Terraform's error outputs to `/terraform-termination-log` to make it available in the containers termination message for better analyzing and more readable error messages (e.g. in the Shoot status).
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,6 +34,38 @@ Apart from dealing with Terraform configuration and state, Terraformer also hand
 signals. It will try to relay `SIGINT` and `SIGTERM` to the running Terraform process in order to stop the ongoing
 infrastructure operations on Pod deletion.
 
+## Termination log
+
+If the Terraform command execution fails (e.g. because of invalid credentials or similar), Terraformer will copy
+Terraform's logs to `$base_dir/terraform-termination-log` (where `$base_dir` is the directory specified in the
+`--base-dir` command line flag (defaults to `/`)). Controllers deploying Terraformer Pods for infrastructure management
+can set `Container.terminationMessagePath` to this file path to have the `kubelet` populate the
```
Comment (rfranzke) [inline README.md:42]:
```suggestion
can set `spec.containers[].terminationMessagePath` in the `Pod` specification to this file path to have the `kubelet` populate the
```

File: README.md [hunk 2]
```diff
@@ -34,6 +34,38 @@ Apart from dealing with Terraform configuration and state, Terraformer also hand
...
+`Pod.status.containerStatuses[].lastState.terminated.message` field (see this [doc](https://kubernetes.io/docs/tasks/debug-application-cluster/determine-reason-pod-failure/#customizing-the-termination-message)
```
Comment (rfranzke) [inline README.md:43]:
```suggestion
`.status.containerStatuses[].lastState.terminated.message` field (see this [doc](https://kubernetes.io/docs/tasks/debug-application-cluster/determine-reason-pod-failure/#customizing-the-termination-message)
```

File: pkg/terraformer/paths.go [hunk 3]
```diff
@@ -23,6 +23,11 @@ type PathSet struct {
 	// ProvidersDir is the directory which contains the provider plugin binaries
 	ProvidersDir string
 
+	// TerminationMessagePath is the file, which the termination log should be written to.
+	// Should be used as Container.terminationMessagePath in the terraformer pod spec,
```
Comment (rfranzke) [inline pkg/terraformer/paths.go:27]:
```suggestion
	// Should be used as spec.containers[].terminationMessagePath in the terraformer pod spec,
```

**Comment by timebertt** [comment]
@rfranzke Done, PTAL.
/test

**Review by rfranzke — APPROVED** [review]
/lgtm

**Comment by timebertt** [comment]
Arg, I was too fast again.
/test

**Pull request merged**